dask/layers.py

```diff
@@ -844,3 +866,82 @@ def _construct_graph(self, deserializing=False):
             dsk[(self.name, i)] = (concat_func, _concat_list)
 
         return dsk
+
+
+class DataFrameIOLayer(Blockwise, DataFrameLayer):
+    """DataFrame-based Blockwise Layer with IO
+
+    Parameters
+    ----------
+    name : string
+        Name to use for the constructed layer.
+    columns : string, list or None
+        Field name(s) to read in as columns in the output.
+    inputs : list[tuple]
+        List of input-argument tuples. Each element will be
+        passed to ``io_func`` to construct a single output
+        partition.
+    io_func : callable
+        A callable function that takes in a single tuple
+        of arguments, and outputs a DataFrame partition.
+    """
+
+    def __init__(
+        self,
+        name,
+        columns,
+        inputs,
+        io_func,
+        part_ids=None,
+        label=None,
+        annotations=None,
+    ):
+        self.name = name
+        self.columns = columns
+        self.inputs = inputs
+        self.io_func = io_func
+        self.part_ids = list(range(len(inputs))) if part_ids is None else part_ids
+        self.label = label
+        self.annotations = annotations
+
+        # Define mapping between key index and "part"
+        io_arg_map = BlockwiseDepDict(
+            {(i,): self.inputs[i] for i in self.part_ids},
+            None,
+        )
+
+        # Use Blockwise initializer
+        dsk = {self.name: (io_func, blockwise_token(0))}
+        super().__init__(
+            self.name,
+            "i",
+            dsk,
+            [(io_arg_map, "i")],
+            {},
+            annotations=annotations,
+        )
```

Comment:
nit: it might help future readers to give all the arguments by name, so you don't have to cross-reference back to `blockwise.py` to figure out what each one means.